### DISCUSSION TITLE
Routing Errors with IP vs Localhost

### DISCUSSION THREAD

**Discussion by GaryJurgens** [quasarframework/quasar#9875] [CLI - Capacitor / Cordova modes] [answered]
hi

I seem to be having a problem,  I have links on my application, Capacitor (android) and web.

if the web is running localhost: with 8080 as the port the links work fine, the URLS look like this

`http://localhost:8080/` (set to Login page)
`http://localhost:8080/register` (goes to register page)

However when its run on mobile it runs from my IP address and the URL looks as follows

`http://192.168.0.103:8080/#/` this loads the login page fine,
but when I click the register link it gives me the Error "Cannot Get /register"

so I entered the URL into my browser and it gave me 

`http://192.168.0.103:8080/#/` - so it loads the Login page

when I clicked on the register link it gives me the same Error, hovering over the it seems to want to go to

**http://192.168.0.103:8080/register**  this results in the same error as above. 

this is the Q button
 ```
<q-btn
            unelevated
            class="text-grey-6"
            type="a"
            href="register" (also /register doesn't work)
            label="Not Registered ? Click Here"
          />
```

this does however work on the "Localhost:8080" 

routing.js

path:'/',
    component:() => import('layouts/PublicLayout.vue'),
    children:[{
    path: '/',
    name:'Login',
    component: () => import('pages/Public/Login.vue')},
    {
      path: 'register',
      name:'Register',
      component: () => import('pages/Public/Register.vue')}]
  },

**Comment by MilosPaunovic** [comment, marked as answer]
@adolfie450  Sounds like you need to configure your server to use history mode for routing.

Check [this](https://quasar.dev/quasar-cli/commands-list#custom-node-server) out for more information.

**Reply by GaryJurgens** [reply]
hi, thanks for the suggestion, in the quasar.conf.json it seems history is enabled.

 // Full list of options: https://v2.quasar.dev/quasar-cli/quasar-conf-js#Property%3A-build
    build: {
      vueRouterMode: 'history', // available values: 'hash', 'history'

**Reply by MilosPaunovic** [reply]
Yup, it's enabled in Quasar itself, but you need to configure your web server separately for it to know how to handle those kind of requests from the browser.

**Reply by GaryJurgens** [reply]
hi thanks i seem to have solved the issue. 

in the quasar.conf.json 

there is the following

history: createHistory(process.env.MODE === 'spa' ? void 0 : process.env.VUE_ROUTER_BASE)

it was originally set to 'ssr'  I changed to 'spa' and seems to do the trick.
